test_projects/test_project_with_custom_main_ut/test_project_with_custom_main_ut.c

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/*the below symbol (CPPUNITTEST_SYMBOL) is defined by testrunnerswitcher.h for the purpose of linking the .dll when compiling as C++. Here there's no testrunnerswitcher so it is defined by hand.*/
+/*it also serves (here) the purpose of not having an empty file*/
+/*this code should be removed when a real project is build from this template*/
+#ifdef CPPUNITTEST_SYMBOL
+#ifdef __cplusplus
+extern "C" {
+#endif
+    void CPPUNITTEST_SYMBOL(void) {}
+
+#ifdef __cplusplus
+}
+#endif
+#endif /*CPPUNITTEST_SYMBOL*/
+
+#include "testrunnerswitcher.h"
+
+BEGIN_TEST_SUITE(test_project_with_custom_main_ut)
+
+TEST_FUNCTION(a)
+{
+
+}
+
```

Comment:
in order to test a custom main that does nothing, would it make sense to have an ASSERT_FAIL() here because this test doesn't actually run (and if it does, it's using the default main which is an error)?

Reply:
er, I guess that probably breaks cppunit...idk just trying to think of how the custom main is actually tested

---
In reply to: [608242494](https://github.com/Azure/c-testrunnerswitcher/pull/66#discussion_r608242494) [](ancestors = 608242494)

Reply:
I made the with_custom_main use a custom main also for CPP. That allows us to test it like you suggested with an ASSERT_FAIL :-).

---
In reply to: [608243861](https://github.com/Azure/c-testrunnerswitcher/pull/66#discussion_r608243861) [](ancestors = 608243861,608242494)